docs/how-to-build-a-lexical-model.md

```diff
@@ -0,0 +1,365 @@
+
+
+### Author
+
+_author_ is an identifier for the maintainer of the model. This
+identifier is a string of one or more ASCII lowercase letters, digits, and the
+underscore (`_`), and it must begin with a lowercase ASCII letter. Some
+organizations have a author identifier already, like `sil` for SIL
+International, `nrc` for the National Research Council Canada, and `fv`
+for FirstVoices.
+
+### BCP 47
+
+_bcp47_ is the [BCP 47][] tag of the language, script, and region you
+are modelling. If you need to uses dashes in your language's BCP 47 tag,
+replace them with underscores; likewise, replace uppercase letters with
+lowercase letters.
+
+Typically, this will just be the [ISO 639-3][] code of your language,
+and nothing else. In the case of SENĆOŦEN, [its ISO 639-3 code is
+`str`](https://iso639-3.sil.org/code/str).
+
+In some cases, you will have to specify the script (writing system), and
+the region. For example, if I'm creating a model for Plains Cree (ISO
+639-3: `crk`) it can be written both in the Latin script ([ISO 15924][]:
+`Latn`) and in Unified Canadian Aboriginal Syllabics ([ISO 15924][]:
+`Cans`). Say my model is only designed for syllabics. The correct BCP 47 tag
+is:
+
+    crk-Cans
+
+However, to create a valid lexical model ID, we need to replace the
+dash with an underscores, and convert all uppercase letters to
+lowercase. Thus, our `bcp47` is:
+
+    crk_cans
+
+[ISO 639-3]: https://iso639-3.sil.org/code_tables/639/data
+[ISO 15924]: https://en.wikipedia.org/wiki/ISO_15924
+[BCP 47]: https://tools.ietf.org/html/bcp47
+
+### Uniq
+
+_uniq_ is an arbitrary tag that uniquely identifies your specific model.
+This identifier can be anything, as long as it hasn't already been used
+before for this specific language. As with `author`
+identifier is one or more ASCII lowercase letters, digits, and the underscore
+(`_`), and it must start with a lowercase ASCII letter.
+
+### Example
+
+My author is called `example`, and the BCP 47 for my language is `str`.
+I'll give my model the unique tag of `tutorial`. Thus my unique model
+is:
+
+    example.str.tutorial
+
+
+Step 4: Creating the package files
+----------------------------------
+
+Now that we have a unique ID for our model, and wordlist data, we can
+the appropriate packaging files.
+
+Say we're currently in a folder called `release`: our model will follow
+this template for its directory structure.
+
+```
+release/
+└── {author}/
+    └── {bcp47}.{uniq}/
+        ├── {author}.{bcp47}.{uniq}.model_info
+        └── source/
+            ├── {author}.{bcp47}.{uniq}.model.kps
+            ├── wordlist.tsv
+            └── model.ts
+```
+
+Notice how under `release`, there is a folder that is named after the
+author. Within that folder is a folder for each models created by that
+particular author. Within each model folder, this is a folder called
+`source` that contains **wordlist.tsv**. We will discuss what the other
+files are called a bit later in this document.
+
```

Comment:
change "each models" to "each model"

"this is a folder..." Did you mean "there is a folder..."?